src/test/evo_deterministicmns_tests.cpp

```diff
@@ -105,6 +105,7 @@ static CMutableTransaction CreateProRegTx(const CChain& active_chain, const CTxM
 
     CProRegTx proTx;
     proTx.nVersion = CProRegTx::GetMaxVersion(!bls::bls_legacy_scheme);
+    proTx.netInfo = MakeNetInfo(proTx);
```

Comment:
(some extra notes related to this line in the same commit)

```
newState->nVersion = opt_proTx->nVersion;
```

version of netInfo should depends on version of CProRegTx, CProUpServTx, etc.

If this object serialized / deserialized - it should not have any "internal version".it should be done only as part of protx / CDeterministicMNState object / etc. They control version of object.

netInfo should no have no any version inside; otherwise I am afraid it would be a bigger nightmare than legacy bls. :)

Reply:
If we need to change ruleset, it means that rules of consensus are changed. It is perfectly matched with protx version bump

Reply:
> the format has been designed with extensibility in mind so hopefully we don't have to revise

Will "unkown" type of data be allowed to get mined in block? I think not, will it?

Reply:
Unknown data is not allowed, they have to meet the _ruleset_ and yes, a ProTx version bump is required in lockstep with the ExtNetInfo bump.